airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/DefaultJobPersistence.java

```diff
@@ -570,14 +579,68 @@ public Optional<Job> getLastSyncJob(final UUID connectionId) throws IOException
         .fetch(BASE_JOB_SELECT_AND_JOIN + WHERE +
             "CAST(jobs.config_type AS VARCHAR) = ? " + AND +
             "scope = ? " +
-            "ORDER BY jobs.created_at DESC LIMIT 1",
+            ORDER_BY_JOB_CREATED_AT_DESC + LIMIT_1,
             Sqls.toSqlName(ConfigType.SYNC),
             connectionId.toString())
         .stream()
         .findFirst()
         .flatMap(r -> getJobOptional(ctx, r.get(JOB_ID, Long.class))));
   }
 
+  /**
+   * For each connection ID in the input, find that connection's latest sync job and return it in an
+   * Optional if one exists.
+   */
+  @Override
+  public List<Optional<Job>> getLastSyncJobsForConnections(final List<UUID> connectionIds) throws IOException {
+    if (connectionIds.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return jobDatabase.query(ctx -> ctx
+        .fetch("SELECT DISTINCT ON (scope) * FROM jobs "
+            + WHERE + "CAST(jobs.config_type AS VARCHAR) = ? "
+            + AND + scopeInList(connectionIds)
+            + "ORDER BY scope, created_at DESC",
+            Sqls.toSqlName(ConfigType.SYNC),
+            connectionIds.stream().map(UUID::toString).map(Names::singleQuote).collect(Collectors.joining(",")))
```

Comment:
can we put this on it's own line and assign to a variable? might just be me - this lines feels very busy today.

Reply:
I actually meant to delete this, I already moved this logic into the `scopeInList` helper but forgot to delete this. There's only one `?` in the query and this is the 2nd argument so it wasn't doing anything 😅